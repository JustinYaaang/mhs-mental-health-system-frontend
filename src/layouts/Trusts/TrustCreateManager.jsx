import React, { Component } from 'react'
// nodejs library to set properties for components
import PropTypes from 'prop-types'
// @material-ui/core components
import withStyles from '@material-ui/core/styles/withStyles'
// core components
import GridItem from 'components/Grid/GridItem.jsx'
import GridContainer from 'components/Grid/GridContainer.jsx'
import PersonForm from 'views/Forms/PersonForm'

const styles = {
  cardCategoryWhite: {
    '&,& a,& a:hover,& a:focus': {
      color: 'rgba(255,255,255,.62)',
      margin: '0',
      fontSize: '14px',
      marginTop: '0',
      marginBottom: '0'
    },
    '& a,& a:hover,& a:focus': {
      color: '#FFFFFF'
    }
  },
  cardTitleWhite: {
    color: '#FFFFFF',
    marginTop: '0px',
    minHeight: 'auto',
    fontWeight: '300',
    fontFamily: "'Roboto', 'Helvetica', 'Arial', sans-serif",
    marginBottom: '3px',
    textDecoration: 'none',
    '& small': {
      color: '#777',
      fontSize: '65%',
      fontWeight: '400',
      lineHeight: '1'
    }
  }
}
/**
 * Component for creating a new Trust Manager
 */
class TrustCreateManager extends Component {
  constructor (props) {
    super(props)
    const { id } = this.props.match.params
    this.state = { id: id
    }
  }



  render () {
    return (
      <GridContainer>
        <GridItem xs={12} sm={12} md={12}>
          <PersonForm hasDetails={false} organization={'TRUSTMANAGER'} />
        </GridItem>
      </GridContainer>
    )
  }
}

TrustCreateManager.propTypes = {
  classes: PropTypes.object
}

export default withStyles(styles)(TrustCreateManager)
