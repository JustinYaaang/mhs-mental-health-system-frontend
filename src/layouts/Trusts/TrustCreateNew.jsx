import React, { Component } from 'react'
// nodejs library to set properties for components
import PropTypes from 'prop-types'
// @material-ui/core components
import withStyles from '@material-ui/core/styles/withStyles'
// core components
import GridItem from 'components/Grid/GridItem.jsx'
import GridContainer from 'components/Grid/GridContainer.jsx'
import Table from 'components/Table/Table.jsx'
import Card from 'components/Card/Card.jsx'
import CardHeader from 'components/Card/CardHeader.jsx'
import CardBody from 'components/Card/CardBody.jsx'
import Tasks from 'components/Tasks/Tasks.jsx'
import AnswerTabs from 'components/CustomTabs/AnswerTabs.jsx'
import TrustServiceFrom from 'routes/node_modules/views/Forms/TrustServiceForm.jsx.js'
import Grade from '@material-ui/icons/Grade'
import Code from '@material-ui/icons/Code'
import { fetchUserAnswers, getPersonel } from '../../services/BackendService'
import { getAnsweredQuestionnaire, getOrganization, getQuestionnaire, getAuthenticationToken, getQuestionnaireWithoutToken, getQuestionnaireWithToken } from '../../services/BackendService'
import TrustServiceForm from 'routes/node_modules/views/Forms/TrustServiceForm.jsx.js'

const styles = {
  cardCategoryWhite: {
    '&,& a,& a:hover,& a:focus': {
      color: 'rgba(255,255,255,.62)',
      margin: '0',
      fontSize: '14px',
      marginTop: '0',
      marginBottom: '0'
    },
    '& a,& a:hover,& a:focus': {
      color: '#FFFFFF'
    }
  },
  cardTitleWhite: {
    color: '#FFFFFF',
    marginTop: '0px',
    minHeight: 'auto',
    fontWeight: '300',
    fontFamily: "'Roboto', 'Helvetica', 'Arial', sans-serif",
    marginBottom: '3px',
    textDecoration: 'none',
    '& small': {
      color: '#777',
      fontSize: '65%',
      fontWeight: '400',
      lineHeight: '1'
    }
  }
}
/**
 * Component for creating a new trust
 */
class TrustCreateNew extends Component {
  constructor (props) {
    super(props)
  }

  componentWillMount () {

  }

  render () {
    return (
      <GridContainer>
        <GridItem xs={12} sm={12} md={12}>
          <TrustServiceForm hasDetails={false} organization={'trust'} />
        </GridItem>
      </GridContainer>
    )
  }
}

TrustCreateNew.propTypes = {
  classes: PropTypes.object
}

export default withStyles(styles)(TrustCreateNew)
