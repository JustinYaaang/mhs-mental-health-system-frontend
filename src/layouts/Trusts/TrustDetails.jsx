import React, { Component } from 'react'
// nodejs library to set properties for components
import PropTypes from 'prop-types'
// @material-ui/core components
import withStyles from '@material-ui/core/styles/withStyles'
// core components
import GridItem from 'components/Grid/GridItem.jsx'
import GridContainer from 'components/Grid/GridContainer.jsx'
import Table from 'components/Table/Table.jsx'
import Card from 'components/Card/Card.jsx'
import CardHeader from 'components/Card/CardHeader.jsx'
import CardBody from 'components/Card/CardBody.jsx'
import Tasks from 'components/Tasks/Tasks.jsx'
import ListRows from 'components/Tasks/ListRows.jsx'
import AnswerTabs from 'components/CustomTabs/AnswerTabs.jsx'
import TrustServiceForm from 'views/Forms/TrustServiceForm.jsx'
import Code from '@material-ui/icons/Code'
import { getPersonnel, deletePersonnel } from '../../services/BackendService'
import swal from 'sweetalert2';
const styles = {
  cardCategoryWhite: {
    '&,& a,& a:hover,& a:focus': {
      color: 'rgba(255,255,255,.62)',
      margin: '0',
      fontSize: '14px',
      marginTop: '0',
      marginBottom: '0'
    },
    '& a,& a:hover,& a:focus': {
      color: '#FFFFFF'
    }
  },
  cardTitleWhite: {
    color: '#FFFFFF',
    marginTop: '0px',
    minHeight: 'auto',
    fontWeight: '300',
    fontFamily: "'Roboto', 'Helvetica', 'Arial', sans-serif",
    marginBottom: '3px',
    textDecoration: 'none',
    '& small': {
      color: '#777',
      fontSize: '65%',
      fontWeight: '400',
      lineHeight: '1'
    }
  }
}

class TrustDetails extends Component {
  constructor (props) {
    super(props)
    const{id}=this.props.match.params //organization's ID
    sessionStorage.setItem('organizationID',id)
    this.state = { id:id,personelList:''
    }
  }

  componentWillMount () {
    getPersonnel().then(response=>{ //Get the personel list from backend
      console.log("response"+ response)
      var counter=1;
      var thelist=new Array() //list for storing the personnel
      response.forEach((map)=>{
        thelist.push([
          counter,map.first_name,map.last_name,map.email,map.trust,map._id
        ])
        counter++
        this.setState({personelList:thelist})
      })
    })



  }

  redirectToManagerDetails = (managerId) => { //Function that redirects to the edit manager page
   this.props.history.push(this.props.history.location.pathname + "/" + managerId)
  }


  createNewUser=()=>{//Function that redirects to the create new manager page 
    this.props.history.push(this.props.history.location.pathname + "/manager/new")

  }

deleteManager=(managerId)=>{
  swal({
    title: "Are you sure?",
    text: "Are you sure you want to delete this entry?",
    icon: "warning",
    buttons: true,
    dangerMode: true,
  })
  .then((willDelete) => {
    if (willDelete) {
      deletePersonnel(managerId).then(response=>{
        swal("The entry has been deleted!", {
          icon: "success",
        });
        this.componentWillMount();
      })
      

    } else {
      swal('Action canceled');
    }
  });











  
}
  render () {
    return (
      <GridContainer>
        <GridItem xs={12} sm={12} md={12}>
          <AnswerTabs
            headerColor='info'
            tabs={[
              {
                tabName: 'DETAILS',
                tabIcon: Code,
                tabContent: (
                    <TrustServiceForm hasDetails={true} organization={"trust"} id={this.state.id} />
                )
              },
              {
                tabName: 'MANAGERS',
                tabIcon: Code,
                tabContent: (
                  <ListRows
                  onDeleteItemClicked={(managerId)=>this.deleteManager(managerId)}
                  createNew={() => this.createNewUser() /*Function for create new manager */}
                     onRowClicked={(managerId) => this.redirectToManagerDetails(managerId)/* Function for edit manager */}
                    tableHeaderColor='primary'
                    tableHead={['S/N', 'First Name', 'Last Name', 'Email', 'Trust Name'] /**Table hearders */}
                    checkedIndexes={[]}
                    tasks={this.state.personelList}
                  />
                )
              }
            ]}
          />
        </GridItem>
      </GridContainer>
    )
  }
}

TrustDetails.propTypes = {
  classes: PropTypes.object
}

export default withStyles(styles)(TrustDetails)
