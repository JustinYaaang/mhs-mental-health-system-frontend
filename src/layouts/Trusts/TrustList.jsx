import React, { Component } from 'react'
// nodejs library to set properties for components
import PropTypes from 'prop-types'
// @material-ui/core components
import withStyles from '@material-ui/core/styles/withStyles'
// core components
import GridItem from 'components/Grid/GridItem.jsx'
import GridContainer from 'components/Grid/GridContainer.jsx'
import AnswerRows from 'components/Tasks/AnswerRows.jsx'
import AnswerTabs from 'components/CustomTabs/AnswerTabs.jsx'
import Code from '@material-ui/icons/Code'
import { getOrganizations,deleteOrganization } from '../../services/BackendService'
import swal from 'sweetalert';
const styles = {
  cardCategoryWhite: {
    '&,& a,& a:hover,& a:focus': {
      color: 'rgba(255,255,255,.62)',
      margin: '0',
      fontSize: '14px',
      marginTop: '0',
      marginBottom: '0'
    },
    '& a,& a:hover,& a:focus': {
      color: '#FFFFFF'
    }
  },
  cardTitleWhite: {
    color: '#FFFFFF',
    marginTop: '0px',
    minHeight: 'auto',
    fontWeight: '300',
    fontFamily: "'Roboto', 'Helvetica', 'Arial', sans-serif",
    marginBottom: '3px',
    textDecoration: 'none',
    '& small': {
      color: '#777',
      fontSize: '65%',
      fontWeight: '400',
      lineHeight: '1'
    }
  }
}

class TrustList extends Component {
  constructor(props) {
    super(props)
    this.state = {
      list: []
    }
  }

  componentWillMount() {
    getOrganizations().then(response=>{
      var i=1;
      var thelist=new
       Array()
      response.forEach((map)=>{
        thelist.push([
          i,map.name,map.description,map.address1+" "+map.address2,map.postcode,map.telephone,map._id
        ])
        i++
        this.setState({list:thelist})
      })
    })
  }

  redirectToTrustDetails = (trustId) => {
    this.props.history.push(this.props.history.location.pathname + "/" + trustId)
  }


  createNewOrganization=()=>{
    this.props.history.push(this.props.history.location.pathname + "/new")
    console.log("!!")
  }

deleteTrust=(trustId)=>{

  swal({
    title: "Are you sure?",
    text: "Are you sure you want to delete this entry?",
    icon: "warning",
    buttons: true,
    dangerMode: true,
  })
  .then((willDelete) => {
    if (willDelete) {
      deleteOrganization(trustId).then(response=>{
        swal("The entry has been deleted!", {
          icon: "success",
        });
        this.componentWillMount();
      })
      

    } else {
      swal('Action canceled');
    }
  });

}

  render() {
    return (
      <GridContainer>
        <GridItem xs={12} sm={12} md={12}>
          <AnswerTabs
            headerColor='info'
            tabs={[
              {
                tabName: 'TRUSTS',
                tabIcon: Code,
                tabContent: (
                  <AnswerRows
                  onDeleteItemClicked={(trustId)=>this.deleteTrust(trustId)}
                    createNew={() => this.createNewOrganization()}
                    onRowClicked={(trustId) => this.redirectToTrustDetails(trustId)}
                    tableHeaderColor='primary'
                    tableHead={['S/N', 'Trust Name', 'Description', 'Address', 'Postcode', 'Telephone']}
                    checkedIndexes={[]}
                    tasks={this.state.list}
                  />
                )
              }
            ]}
          />
        </GridItem>
      </GridContainer>
    )
  }
}

TrustList.propTypes = {
  classes: PropTypes.object
}

export default withStyles(styles)(TrustList)
