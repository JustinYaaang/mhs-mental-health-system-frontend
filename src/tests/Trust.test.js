import React from 'react'
import { shallow } from 'enzyme'
import TrustDetails from '../layouts/Trusts/TrustDetails'
import TrustList from '../layouts/Trusts/TrustList'
import TrustCreateManager from '../layouts/Trusts/TrustCreateManager'
import TrustCreateNew from '../layouts/Trusts/TrustCreateNew'
import TrustUserDetails from '../layouts/Trusts/TrustUserDetails'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

describe('TrustDetails', () => {
  it('should render correctly', async () => {
     const component = shallow(<TrustDetails />)

     expect(component).toMatchSnapshot()
  })
})
describe('TrustList', () => {
    it('should render correctly', async () => {
       const component = shallow(<TrustList />)
  
       expect(component).toMatchSnapshot()
    })
  })

  describe('Trust Create Manager', () => {
   it('should render correctly', async () => {
      const component = shallow(<TrustCreateManager />)
 
      expect(component).toMatchSnapshot()
   })
 })
 describe('Trust Create New', () => {
   it('should render correctly', async () => {
      const component = shallow(<TrustCreateNew />)
 
      expect(component).toMatchSnapshot()
   })
 })
 describe('Trust User Details', () => {
   it('should render correctly', async () => {
      const component = shallow(<TrustUserDetails />)
 
      expect(component).toMatchSnapshot()
   })
 })
