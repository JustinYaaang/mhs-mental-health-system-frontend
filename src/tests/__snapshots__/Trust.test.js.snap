// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trust Create Manager should render correctly 1`] = `
<TrustCreateManager
  classes={
    Object {
      "cardCategoryWhite": "TrustCreateManager-cardCategoryWhite-5",
      "cardTitleWhite": "TrustCreateManager-cardTitleWhite-6",
    }
  }
/>
`;

exports[`Trust Create New should render correctly 1`] = `
<TrustCreateNew
  classes={
    Object {
      "cardCategoryWhite": "TrustCreateNew-cardCategoryWhite-7",
      "cardTitleWhite": "TrustCreateNew-cardTitleWhite-8",
    }
  }
/>
`;

exports[`Trust User Details should render correctly 1`] = `
<UserDetails
  classes={
    Object {
      "cardCategoryWhite": "UserDetails-cardCategoryWhite-9",
      "cardTitleWhite": "UserDetails-cardTitleWhite-10",
    }
  }
/>
`;

exports[`TrustDetails should render correctly 1`] = `
<TrustDetails
  classes={
    Object {
      "cardCategoryWhite": "TrustDetails-cardCategoryWhite-1",
      "cardTitleWhite": "TrustDetails-cardTitleWhite-2",
    }
  }
/>
`;

exports[`TrustList should render correctly 1`] = `
<TrustList
  classes={
    Object {
      "cardCategoryWhite": "TrustList-cardCategoryWhite-3",
      "cardTitleWhite": "TrustList-cardTitleWhite-4",
    }
  }
/>
`;
